impl/src/main/java/org/jboss/weld/util/DeploymentStructures.java

```diff
@@ -49,4 +54,23 @@ public static BeanDeployment getOrCreateBeanDeployment(Deployment deployment, Be
         }
     }
 
+    public static <E> BeanDeployment getOrCreateBeanDeploymentForExtension(Deployment deployment,
```

Comment:
Hm, why do we place this logic in the core? Couldn't we just modify the logic in  `org.jboss.weld.environment.deployment.WeldDeployment.loadBeanDeploymentArchive()`? I mean we have the extension metadata in the `org.jboss.weld.environment.se.Weld` so we could extend the deployment created in the `Weld.createDeployment()` method or even use an anonymous class like:
```java
deployment = new WeldDeployment(resourceLoader, bootstrap, beanDeploymentArchives, extensions) {
      @Override
       public BeanDeploymentArchive loadBeanDeploymentArchive(Class<?> beanClass) {
           // do something SE-specific...
        }
};
```

Reply:
Hmm, I like the idea of anonymous class, I'll try that!

Reply:
@mkouba I've reworked it to use the anonymous class and it does look better, thanks!
Please re-review when you have a spare minute.